packages/block-library/src/categories/index.php

```diff
@@ -16,11 +16,6 @@ function render_block_core_categories( $attributes ) {
 	static $block_id = 0;
 	$block_id++;
 
-	$align = 'center';
-	if ( isset( $attributes['align'] ) && in_array( $attributes['align'], array( 'left', 'right', 'full' ), true ) ) {
```

Comment:
I agree we shouldn't want to recreate validation in the render callback. Ideally this is validated at the attribute definition, i.e. [via `enum`](https://json-schema.org/understanding-json-schema/reference/generic.html#enumerated-values).

I expect this is encompassed in part by #2751.

Separately, we can plan to support `enum` validation in the client implementation. Maybe a near-term compromise is adding expected alignments as an `enum` in the attribute definition to be ready when this is "turned on". Though, ideally this would be handled automatically when opting into align supports, not the responsibility of each block implementer to define.

Reply:
The `register_block_type` PHP function can accept the same options as what exists in the client today. They are passed down and shallowly† merged with client registration. With #2751, the idea is that the server would be the primary location where blocks be registered.

† All of `attributes` must be defined in the server or in the client, not fragments amongst both.

Reply:
My hope is to actually kill off `getEditWrapperProps` ([example](https://github.com/WordPress/gutenberg/pull/7841#issuecomment-406319849)).

I don't think we want to be validating there either. As-is, I think it'd be sufficient to just pass through any non-`undefined` as `data-align`, not too much unlike how we've done with the frontend class rendering here?

Reply:
@aduth Alright then, I have created a new PR to do just that and move the attributes from JS to PHP for the Categories block: #10635.